git.md

```diff
@@ -87,6 +87,22 @@ $ git reset --soft HEAD^
 $ git rebase --continue  # as before
 ```
 
+### Common Pitfall - Inclusion of unwanted commits from a rebase
+If you included commits you didn't want to, this [Q&A on Stackoverflow](https://stackoverflow.com/questions/134882/undoing-a-git-rebase) provide a solution. Example:
+
+Using reflog to find the head commit of the branch as it was immediatly before the rebase started.
+```
+git reflog
+```
+and to reset the current branch to it (with the usual caveats about being absolutely sure before reseting with the --hard option). X will be the one you reset to.
```

Comment:
code highlight `--hard`.

on the other note, if this is a copy/paste from the source Q&A, perhaps omitting the entire thing is cleaner.